lib/json_api_client/errors.rb

```diff
@@ -1,10 +1,31 @@
+require 'rack'
+
 module JsonApiClient
   module Errors
     class ApiError < StandardError
       attr_reader :env
+
       def initialize(env, msg = nil)
-        super msg
         @env = env
+        # Try to fetch json_api errors from response
+        msg = track_json_api_errors(msg)
+
+        super msg
+      end
+
+      private
+
+      # Try to fetch json_api errors from response
+      def track_json_api_errors(msg)
+        return msg unless env.try(:body).kind_of?(Hash) || env.body.key?('errors')
+
+        errors_msg = env.body['errors'].map { |e| e['title'] }.compact.join('; ').presence
```

Comment:
I was about to start work on something similar to this, because I ran into issues of one rails app getting 400 errors from another, and what got logged to my error reporting service wasn't very helpful.

This looks really excellent to me! What do you think about adding a configuration that would change the messages here to either get this simplified error reporting you've added or to make it dump the entire response body for more information?

If we think that's a good idea, but @stokarenko doesn't want to or can't take the time to implement it, I could definitely make an additional PR to add that once this is merged.

Reply:
Ooh! I like this config idea a lot. @stokarenko let me know if you're going to add the config. If not, i'll merge it in and push the new version.